cloud/amazon/s3.py

```diff
@@ -248,7 +248,7 @@
 '''
 
 import os
-import urlparse
+from six.moves.urllib.parse import urlparse
```

Comment:
Rather than depend on six being installed on the target system we use a bundled copy.  So you can do this as 
``` python
from ansible.module_utils.six.moves.urllib.parse import urlparse
```